proxy.go

```diff
@@ -211,7 +211,12 @@ func route(w http.ResponseWriter, r *http.Request) {
 	excludedHosts := newSet()
 	hosts, version, excludedRegions := browsers.find(browser, version, excludedHosts, newSet())
 	confLock.RUnlock()
-	if len(hosts) == 0 {
+
+	if(len(hosts) ==0 && user == guestUserName ) {
+		reply(w, errMsg("Guest access is unavailable."), http.StatusUnauthorized)
+		log.Printf("[%d] [UNSUPPORTED_BROWSER] [%s] [%s] [%s]\n", id, user, remote, fmtBrowser(browser, version))
+		return
+	} else if len(hosts) == 0 {
```

Comment:
I think we should return 401 in `WithSuitableAuthentication` instead. You can check whether guest quota exists like the following:
```
confLock.RLock()
if _, guestQuotaExists := quota[user]; !guestQuotaExists {
    //Return 401
}
confLock.RUnlock()
```